### PR TITLE
🎨 Add Snackbar view

### LINKED ISSUE #81: Implement a Snackbar mechanism
## Description
Since iOS does not have default snackbars, we should implement our own. As discussed [here](https://github.com/toggl/mobile-discussions/issues/59), most of the team agrees on making something similar to what we already have on Xamarin (but make it more Swift-y). Let's do so and ask for @Odrakir's blessing.

I think a good change we can do is make sure that the snackbar is bound to a specific kind of action? Something like: `Snackbar<TimeEntriesLogAction>.from(store, .undoButtonTapped).show()` sounds reasonable enough. 

## Definition of done
- [ ] Some method that displays a snackbar is added to the codebase
- [ ] Calling said method makes the snackbar appear for 5 seconds and then disappear
- [ ] The snackbar method has an undo button that is wired to emitting some action

### PR DESCRIPTION
**Pull request by juanlaube**
This adds the Snackbar view to the Utils module. The API is similar to Android's Snackbar for consistency.
I also added a new module, UIUtils, that depends on Utils and Architecture. This module adds an extension method on the Snackbar to use it with a store.

Closes #81

### CODE REVIEW

File: Modules/Utils/Utils/UI/Snackbar.swift [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+public final class Snackbar: UIView {
+
+    private let height: CGFloat = 64
+    private let horizontalPadding: CGFloat = 20
+    private let verticalPadding: CGFloat = 10
+    private let verticalPaddingInWindow: CGFloat = 40
+    private let animationDuration: TimeInterval = 0.3
+
+    public enum Duration {
+        case short
```
Comment (heytherewill) [inline Modules/Utils/Utils/UI/Snackbar.swift:17]: Given that our snackbars only have one possible duration, I don't think this Duration enum is even needed

File: Modules/Utils/Utils/UI/Snackbar.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        
+        var duration: TimeInterval {
+            switch self {
+            case .short: return 5
```
Comment (heytherewill) [inline Modules/Utils/Utils/UI/Snackbar.swift:21]: Do we have a constants file? I'd move this there if possible

File: Modules/Utils/Utils/UI/Snackbar.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import UIKit
+
+public protocol SnackbarHost {
+    var snackbarHostBottomAnchor: NSLayoutYAxisAnchor { get }
+}
+
+public final class Snackbar: UIView {
+
+    private let height: CGFloat = 64
+    private let horizontalPadding: CGFloat = 20
```
Comment (heytherewill) [inline Modules/Utils/Utils/UI/Snackbar.swift:11]: The problem of having paddings here is that if we ever decide to change the padding of the views displaying the toasts, we'd need to remember and update this file, which is not ideal

File: Modules/Utils/Utils/UI/Snackbar.swift (hunk 2 above)
Comment (juanlaube) [reply Modules/Utils/Utils/UI/Snackbar.swift:21]: We don't have one, but I made it a constant in the Snackbar class.

File: Modules/Utils/Utils/UI/Snackbar.swift (hunk 3 above)
Comment (juanlaube) [reply Modules/Utils/Utils/UI/Snackbar.swift]: Paddings are gone 🔥 
I used the view controller's layout margins.

File: Modules/UIUtils/UIUtils/UIUtils.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+//
+//  UIUtils.h
+//  UIUtils
+//
+//  Created by Juan Laube on 4/2/20.
+//  Copyright © 2020 Toggl Track. All rights reserved.
+//
+
```
Comment (heytherewill) [inline Modules/UIUtils/UIUtils/UIUtils.h:8]: 🔥 ?

File: Modules/UIUtils/UIUtilsTests/UIUtilsTests.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
+//
+//  UIUtilsTests.swift
+//  UIUtilsTests
+//
+//  Created by Juan Laube on 4/2/20.
+//  Copyright © 2020 Toggl Track. All rights reserved.
+//
+
```
Comment (heytherewill) [inline Modules/UIUtils/UIUtilsTests/UIUtilsTests.swift:8]: 🔥

File: Modules/UIUtils/UIUtils/UIUtils.h (hunk 4 above)
Comment (juanlaube) [reply Modules/UIUtils/UIUtils/UIUtils.h]: Stupid Xcode 🤦‍♂

File: Modules/UIUtils/UIUtilsTests/UIUtilsTests.swift (hunk 5 above)
Comment (juanlaube) [reply Modules/UIUtils/UIUtilsTests/UIUtilsTests.swift]: Same

**Comment by Odrakir** [comment]
By the way, why is the checksum in the Podfile.lock changing?

**Pull request merged**